src/topology/sober.lean

```diff
@@ -0,0 +1,282 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.separation
+import topology.continuous_function.basic
+
+/-!
+# Sober spaces
+
+A sober space is a topological space where every irreducible closed subset has a generic point.
+Note that we do not require the generic point to be unique.
+
+## Main definition
+
+* `specializes` : `specializes x y` (`x ⤳ y`) means that `x` specializes to `y`, i.e.
+  `y` is in the closure of `x`.
+* `specialization_preorder` : specialization gives a preorder on the topological space.
+* `specialization_order` : specialization gives a preorder on the topological space.
+* `is_generic_point` : `x` is the generic point of `S` if `S` is the closure of `x`.
+* `sober_space` : A space is sober if every irreducible closed subset has a generic point.
+
+-/
+
+variables {α β : Type*} [topological_space α] [topological_space β]
+
+section specialize_order
+
+/-- `x` specializes to `y` if `y` is in the closure of `x`. The notation used is `x ⤳ y`. -/
+def specializes (x y : α) : Prop := y ∈ closure ({x} : set α)
+
+infix ` ⤳ `:300 := specializes
```

Comment:
Is there already an abbreviation in VScode to type this symbol? I don't think so, right? So we should have an accompanying PR to https://github.com/leanprover/vscode-lean/blob/master/src/abbreviation/abbreviations.json

Reply:
Added at https://github.com/leanprover/vscode-lean/pull/285